### ISSUE TITLE
Trigger Redraw Crash

### ISSUE THREAD

**Issue by shmuelie** [microsoft/terminal#18275]
### Windows Terminal version

1.21.3231.0

### Windows build number

10.0.26335.0

### Steps to reproduce

Unsure 😅

### Expected Behavior

To not crash

### Actual Behavior

`Access Violation` exception, see Watson Bucket 20f45815-12a1-7b5a-eadf-5a6297346f30

**Comment by zadjii-msft** [comment]
internal dupe chain leads to MSFT:54447014, 

`ACCESS_VIOLATION_c0000005_Microsoft.Terminal.Control.dll!
Microsoft::Console::VirtualTerminal::StateMachine::ProcessString

**Comment by zadjii-msft** [comment]
and no hits for that on 1.22. @shmuelie  if you use Terminal Preview, does this go away?

**Comment by lhecker** [comment]
Ah fuck I figured it out. And I typed out such a long message before that...

Here's where we destroy the renderer:
https://github.com/microsoft/terminal/blob/bfab5fde4d09ae138b9a26c8fd8fb071d5f5f193/src/cascadia/TerminalControl/ControlCore.cpp#L227-L233

But during the problematic call we only hold a strong reference to the `Terminal`, not the `ControlCore`:
https://github.com/microsoft/terminal/blob/bfab5fde4d09ae138b9a26c8fd8fb071d5f5f193/src/cascadia/TerminalControl/ControlCore.cpp#L197-L201

In other words, the `ControlCore` gets destroyed while we're holding onto the dysfunctional carcass of a terminal.

**Comment by shmuelie** [comment]
> and no hits for that on 1.22. [@shmuelie](https://github.com/shmuelie) if you use Terminal Preview, does this go away?

I've not been able to reproduce it since the first crash so can't say

**Comment by shmuelie** [comment]
Just hit it again today on 1.23.3582.0

**Comment by lhecker** [comment]
I think this may have been fixed by #18632.
